test/test_map_processor_detection_oils_example.py

```diff
@@ -52,6 +54,38 @@ def test_map_processor_detection_oil_example():
 
     map_processor.run()
 
+def test_map_processor_detection_oil_example_using_cache():
+    qgs = init_qgis()
+    rlayer = create_rlayer_from_file(RASTER_FILE_PATH)
+
+    model_wrapper = Detector(MODEL_FILE_PATH)
+
+    params = DetectionParameters(
+        resolution_cm_per_px=150,
+        tile_size_px=model_wrapper.get_input_size_in_pixels()[0],  # same x and y dimensions, so take x
+        batch_size=1,
+        local_cache=True,
+        processed_area_type=ProcessedAreaType.ENTIRE_LAYER,
+        mask_layer_id=None,
+        input_layer_id=rlayer.id(),
+        input_channels_mapping=INPUT_CHANNELS_MAPPING,
+        processing_overlap=ProcessingOverlap(ProcessingOverlapOptions.OVERLAP_IN_PERCENT, percentage=40),
+        model=model_wrapper,
+        confidence=0.5,
+        iou_threshold=0.1,
+        remove_overlapping_detections=False,
+        model_output_format=ModelOutputFormat.ALL_CLASSES_AS_SEPARATE_LAYERS,
+        model_output_format__single_class_number=-1,
+    )
+
+    map_processor = MapProcessorDetection(
+        rlayer=rlayer,
+        vlayer_mask=None,
+        map_canvas=MagicMock(),
+        params=params,
+    )
+
+    map_processor.run()
```

Comment:
How about adding an assert for the result (also in the original non-batched test)